.github/workflows/update.yml

```diff
@@ -11,42 +11,88 @@ jobs:
 
     runs-on: ubuntu-latest
     name: Upstream cucumber update
+    permissions:
+      contents: read
+      pull-requests: read
     steps:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: 8.1
+          php-version: 8.4
           coverage: none
 
-      - uses: actions/checkout@v4
+      - name: Check whether to update or create PR
+        id: check-existing-pr
+        uses: actions/github-script@v7
+        with:
+          script: |
+            async function findOpenPR({owner, repo, branchPrefix}) {
+              const iterator = github.paginate.iterator(github.rest.pulls.list, {owner, repo, state: 'open'})
+              const repoFullName = `${owner}/${repo}`
 
-      - name: Install dependencies
-        run: composer update
+              for await (const response of iterator) {
+                const matchingPR = response.data.find((pr) => (
+                  pr.head.ref.startsWith(branchPrefix)
+                  && (pr.head.repo.full_name === repoFullName)
+                ))
 
-      - name: Update cucumber tag
-        id: cucumber
-        run: bin/update_cucumber
+                if (matchingPR) {
+                  return matchingPR
+                }
+              }
+
+              // No existing PRs
+              return false
+            }
 
-      - name: Re-install dependencies
-        run: composer update
-        if: steps.cucumber.outputs.cucumber_version
+            const branchPrefix = 'cucumber-update-'
 
-      - name: Update translations
-        run: bin/update_i18n
-        if: steps.cucumber.outputs.cucumber_version
+            const pr = await findOpenPR({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              branchPrefix
+            })
 
-      - name: Find changelog
-        id: changelog
-        run: bin/cucumber_changelog ${{ steps.cucumber.outputs.cucumber_version }}
-        if: steps.cucumber.outputs.cucumber_version
+            if (pr) {
```

Comment:
We first check to see if there's an existing `cucumber-update-*` branch in this repo with an open PR. If there is, that will be checked out and the next cucumber update (if any) applied to the same branch.

If not, we create a new branch based off master. We use a unique suffix on the branch name to avoid any conflict with an old update (if e.g. a previous PR was merged/closed but the branch was not deleted).